### PR TITLE
Amend documents for archival

### PR DESCRIPTION
**Pull request by yujunz**
* principals are not mandatory for role policy
* typo of enableRead
* clarify the service account to trust

### CODE REVIEW

File: docs/features/archival.md [hunk 1]
```diff
@@ -55,7 +51,7 @@ Then create a trust relationships for your EKS cluster:
             "Action": "sts:AssumeRoleWithWebIdentity",
             "Condition": {
                 "StringEquals": {
-                    "oidc.eks.<aws_region>.amazonaws.com/id/<cluster_id>:sub": ["system:serviceaccount:<temporal_ns>:<temporal_sa>"]
+                    "oidc.eks.<aws_region>.amazonaws.com/id/<cluster_id>:sub": ["system:serviceaccount:<temporal_ns>:<temporal_history_sa>"]
```
Comment (yujunz) [inline docs/features/archival.md:54]: Not very sure about this. Please double check.

Comment (alexandrevilain) [reply docs/features/archival.md:54]: This is the same SA for all services, so no need to specify "history"

Comment (yujunz) [reply docs/features/archival.md:54]: Is it required by all service account?

Comment (yujunz) [reply docs/features/archival.md:54]: It seems each service has its own service account.

```
kubectl get sa -n temporal-tok
NAME                    SECRETS   AGE
default                 0         19h
temporal-tok-frontend   0         61m
temporal-tok-history    0         61m
temporal-tok-matching   0         61m
temporal-tok-worker     0         61m
```

Comment (alexandrevilain) [reply docs/features/archival.md:54]: Sorry, answered to quickly! You're right, there is one SA per service! 
All services have the `eks.amazonaws.com/role-arn` annotation. I don't know which service is responsible to connect to s3 for archival. I'll try to find the info!

Comment (DodgeCamaro) [reply docs/features/archival.md:54]: @yujunz  @alexandrevilain  History service responsible for archive workflow

Comment (DodgeCamaro) [reply docs/features/archival.md:54]: image

Comment (DodgeCamaro) [reply docs/features/archival.md:54]: at feature release, need to add worker SA if use aws IAM access to OpenSearch/RDS

Comment (alexandrevilain) [reply docs/features/archival.md:54]: So let's merge this PR, thanks @DodgeCamaro !

**Review by alexandrevilain — APPROVED** [review]

**Pull request merged**